### ISSUE TITLE
Unable to build server from local dev box or from peertube server

### ISSUE THREAD

**Issue by FlatEarthTruther** [Chocobozzz/PeerTube#1171]
I've tried to build the server code both on my mac and on the debian 9 server that is currently running 1.0.0-beta-15 without success. It fails on the migrations part.

```
> peertube@1.0.0-beta.16 build /var/www/peertube/PeerTube
> SCRIPTY_PARALLEL=true scripty


> peertube@1.0.0-beta.16 tsc /var/www/peertube/PeerTube
> tsc


> peertube-client@1.0.0-beta.16 ng /var/www/peertube/PeerTube/client
> ng "build" "--output-path" "dist/en_US/" "--deploy-url" "/client/en_US/" "--prod" "--stats-json"

 10% building modules 3/3 modules 0 active(node:25411) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
 69% building modules 813/814 modules 1 active ...aption-add-modal.component.ngfactory.jsserver/controllers/api/videos/import.ts:219:3 - error TS2322: Type 'Bluebird<any>' is not assignable to type 'Bluebird<VideoImportModel>'.
  Types of property 'then' are incompatible.
    Type '{ <U>(onFulfill?: (value: any) => Resolvable<U>, onReject?: (error: any) => Resolvable<U>): Blueb...' is not assignable to type '{ <U>(onFulfill?: (value: VideoImportModel) => U | PromiseLike<U>, onReject?: (error: any) => U |...'.
      Type 'Bluebird<any>' is not assignable to type 'Bluebird<any>'. Two different types with this name exist, but they are unrelated.

219   return sequelizeTypescript.transaction(async t => {
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
220     const sequelizeOptions = { transaction: t }
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
...
243     return videoImport
    ~~~~~~~~~~~~~~~~~~~~~~
244   })
    ~~~~


server/initializers/migrations/0005-email-pod.ts:18:3 - error TS2719: Type 'Bluebird<void>' is not assignable to type 'Bluebird<void>'. Two different types with this name exist, but they are unrelated.
  Types of property 'then' are incompatible.
    Type '{ <U>(onFulfill?: (value: void) => Resolvable<U>, onReject?: (error: any) => Resolvable<U>): Blue...' is not assignable to type '{ <U>(onFulfill?: (value: void) => U | PromiseLike<U>, onReject?: (error: any) => U | PromiseLike...'.
      Type 'Bluebird<any>' is not assignable to type 'Bluebird<any>'. Two different types with this name exist, but they are unrelated.

 18   return q.addColumn('Pods', 'email', data)
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 19     .then(() => {
    ~~~~~~~~~~~~~~~~~
...
 26       return q.changeColumn('Pods', 'email', data)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 27     })
    ~~~~~~


server/initializers/migrations/0010-email-user.ts:17:3 - error TS2719: Type 'Bluebird<void>' is not assignable to type 'Bluebird<void>'. Two different types with this name exist, but they are unrelated.

 17   return q.addColumn('Users', 'email', data)
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 18     .then(() => {
    ~~~~~~~~~~~~~~~~~
...
 25       return q.changeColumn('Users', 'email', data)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 26     })
    ~~~~~~


server/initializers/migrations/0015-video-views.ts:17:3 - error TS2719: Type 'Bluebird<void>' is not assignable to type 'Bluebird<void>'. Two different types with this name exist, but they are unrelated.

17   return q.addColumn('Videos', 'views', data)
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


server/initializers/migrations/0020-video-likes.ts:17:3 - error TS2719: Type 'Bluebird<void>' is not assignable to type 'Bluebird<void>'. Two different types with this name exist, but they are unrelated.

17   return q.addColumn('Videos', 'likes', data)
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


server/initializers/migrations/0025-video-dislikes.ts:17:3 - error TS2719: Type 'Bluebird<void>' is not assignable to type 'Bluebird<void>'. Two different types with this name exist, but they are unrelated.

17   return q.addColumn('Videos', 'dislikes', data)
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


server/initializers/migrations/0030-video-category.ts:18:3 - error TS2719: Type 'Bluebird<void>' is not assignable to type 'Bluebird<void>'. Two different types with this name exist, but they are unrelated.

 18   return q.addColumn('Videos', 'category', data)
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 19     .then(() => {
    ~~~~~~~~~~~~~~~~~
...
 22       return q.changeColumn('Videos', 'category', data)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 23     })
    ~~~~~~


server/initializers/migrations/0035-video-licence.ts:18:3 - error TS2719: Type 'Bluebird<void>' is not assignable to type 'Bluebird<void>'. Two different types with this name exist, but they are unrelated.

 18   return q.addColumn('Videos', 'licence', data)
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 19     .then(() => {
    ~~~~~~~~~~~~~~~~~
...
 21       return q.changeColumn('Videos', 'licence', data)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 22     })
    ~~~~~~


server/initializers/migrations/0040-video-nsfw.ts:18:3 - error TS2719: Type 'Bluebird<void>' is not assignable to type 'Bluebird<void>'. Two different types with this name exist, but they are unrelated.

 18   return q.addColumn('Videos', 'nsfw', data)
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 19     .then(() => {
    ~~~~~~~~~~~~~~~~~
...
 22       return q.changeColumn('Videos', 'nsfw', data)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 23     })
    ~~~~~~


server/initializers/migrations/0045-user-display-nsfw.ts:17:3 - error TS2719: Type 'Bluebird<void>' is not assignable to type 'Bluebird<void>'. Two different types with this name exist, but they are unrelated.

17   return q.addColumn('Users', 'displayNSFW', data)
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


server/initializers/migrations/0050-video-language.ts:17:3 - error TS2719: Type 'Bluebird<void>' is not assignable to type 'Bluebird<void>'. Two different types with this name exist, but they are unrelated.

17   return q.addColumn('Videos', 'language', data)
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


server/initializers/migrations/0055-video-uuid.ts:18:3 - error TS2719: Type 'Bluebird<void>' is not assignable to type 'Bluebird<void>'. Two different types with this name exist, but they are unrelated.

 18   return q.addColumn('Videos', 'uuid', dataUUID)
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 19     .then(() => {
    ~~~~~~~~~~~~~~~~~
...
102       return q.removeColumn('Videos', 'remoteId')
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
103     })
    ~~~~~~


server/initializers/migrations/0070-user-video-quota.ts:19:3 - error TS2719: Type 'Bluebird<void>' is not assignable to type 'Bluebird<void>'. Two different types with this name exist, but they are unrelated.

 19   return q.addColumn('Users', 'videoQuota', data)
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 20     .then(() => {
    ~~~~~~~~~~~~~~~~~
...
 22       return q.changeColumn('Users', 'videoQuota', data)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 23     })
    ~~~~~~


server/initializers/migrations/0130-user-autoplay-video.ts:17:3 - error TS2719: Type 'Bluebird<void>' is not assignable to type 'Bluebird<void>'. Two different types with this name exist, but they are unrelated.

17   return q.addColumn('user', 'autoPlayVideo', data)
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


server/lib/activitypub/actor.ts:228:3 - error TS2322: Type 'Bluebird<any>' is not assignable to type 'Promise<ActorModel> | Bluebird<ActorModel>'.
  Type 'Bluebird<any>' is not assignable to type 'Bluebird<ActorModel>'.
    Types of property 'then' are incompatible.
      Type '{ <U>(onFulfill?: (value: any) => Resolvable<U>, onReject?: (error: any) => Resolvable<U>): Blueb...' is not assignable to type '{ <U>(onFulfill?: (value: ActorModel) => U | PromiseLike<U>, onReject?: (error: any) => U | Promi...'.
        Type 'Bluebird<any>' is not assignable to type 'Bluebird<any>'. Two different types with this name exist, but they are unrelated.

228   return sequelizeTypescript.transaction(t => save(t))
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


server/lib/activitypub/process/process-delete.ts:35:38 - error TS2345: Argument of type '(byActor: ActorModel, videoComment: VideoCommentModel, activity: ActivityDelete) => Bluebird<any>' is not assignable to parameter of type '(arg1: ActorModel, arg2: VideoCommentModel, arg3: ActivityDelete) => Promise<any> | Bluebird<any>'.
  Type 'Bluebird<any>' is not assignable to type 'Promise<any> | Bluebird<any>'.
    Type 'Bluebird<any>' is not assignable to type 'Bluebird<any>'. Two different types with this name exist, but they are unrelated.

35       return retryTransactionWrapper(processDeleteVideoComment, byActor, videoCommentInstance, activity)
                                        ~~~~~~~~~~~~~~~~~~~~~~~~~


server/lib/activitypub/process/process-undo.ts:30:36 - error TS2345: Argument of type '(follower: ActorModel, followActivity: ActivityFollow) => Bluebird<any>' is not assignable to parameter of type '(arg1: ActorModel, arg2: ActivityFollow) => Promise<any> | Bluebird<any>'.
  Type 'Bluebird<any>' is not assignable to type 'Promise<any> | Bluebird<any>'.
    Type 'Bluebird<any>' is not assignable to type 'Bluebird<any>'. Two different types with this name exist, but they are unrelated.

30     return retryTransactionWrapper(processUndoFollow, byActor, activityToUndo)
                                      ~~~~~~~~~~~~~~~~~


server/lib/activitypub/process/process-undo.ts:34:36 - error TS2345: Argument of type '(byActor: ActorModel, announceActivity: ActivityAnnounce) => Bluebird<any>' is not assignable to parameter of type '(arg1: ActorModel, arg2: ActivityAnnounce) => Promise<any> | Bluebird<any>'.
  Type 'Bluebird<any>' is not assignable to type 'Promise<any> | Bluebird<any>'.
    Type 'Bluebird<any>' is not assignable to type 'Bluebird<any>'. Two different types with this name exist, but they are unrelated.

34     return retryTransactionWrapper(processUndoAnnounce, byActor, activityToUndo)
                                      ~~~~~~~~~~~~~~~~~~~


server/lib/avatar.ts:17:34 - error TS2345: Argument of type '() => Bluebird<any>' is not assignable to parameter of type '() => Promise<any> | Bluebird<any>'.
  Type 'Bluebird<any>' is not assignable to type 'Promise<any> | Bluebird<any>'.
    Type 'Bluebird<any>' is not assignable to type 'Bluebird<any>'. Two different types with this name exist, but they are unrelated.

17   return retryTransactionWrapper(() => {
                                    ~~~~~~~


server/lib/job-queue/handlers/activitypub-follow.ts:35:34 - error TS2345: Argument of type '(fromActor: ActorModel, targetActor: ActorModel) => Bluebird<any>' is not assignable to parameter of type '(arg1: ActorModel, arg2: ActorModel) => Promise<any> | Bluebird<any>'.
  Type 'Bluebird<any>' is not assignable to type 'Promise<any> | Bluebird<any>'.
    Type 'Bluebird<any>' is not assignable to type 'Bluebird<any>'. Two different types with this name exist, but they are unrelated.

35   return retryTransactionWrapper(follow, fromActor, targetActor)
                                    ~~~~~~


server/models/activitypub/actor-follow.ts:489:18 - error TS2345: Argument of type 'Bluebird<any>' is not assignable to parameter of type 'Bluebird<any>'.

489       tasks.push(ActorFollowModel.sequelize.query(query, options))
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


server/models/video/tag.ts:78:5 - error TS2322: Type 'Bluebird<any>' is not assignable to type 'Bluebird<string[]>'.
  Types of property 'then' are incompatible.
    Type '{ <U>(onFulfill?: (value: any) => Resolvable<U>, onReject?: (error: any) => Resolvable<U>): Blueb...' is not assignable to type '{ <U>(onFulfill?: (value: string[]) => U | PromiseLike<U>, onReject?: (error: any) => U | Promise...'.
      Type 'Bluebird<any>' is not assignable to type 'Bluebird<any>'. Two different types with this name exist, but they are unrelated.
        Types of property 'error' are incompatible.
          Type '<U>(onReject: (reason: any) => Resolvable<U>) => Bluebird<U>' is not assignable to type '<U>(onReject: (reason: any) => U | PromiseLike<U>) => Bluebird<U>'.
            Type 'Bluebird<U>' is not assignable to type 'Bluebird<U>'. Two different types with this name exist, but they are unrelated.
              Types of property 'catch' are incompatible.
                Type '{ <U = U>(onReject: (error: any) => Resolvable<U>): Bluebird<U | U>; <E1, E2, E3, E4, E5>(filter1...' is not assignable to type '{ (onReject: (error: any) => U | PromiseLike<U>): Bluebird<U>; <U>(onReject: (error: any) => U | ...'.
                  Type 'Bluebird<any>' is not assignable to type 'Bluebird<U>'.
                    Types of property 'map' are incompatible.
                      Type '<U>(mapper: IterateFunction<{}, U>, options?: ConcurrencyOption) => Bluebird<U[]>' is not assignable to type '<Q, U>(mapper: (item: Q, index: number, arrayLength: number) => U | PromiseLike<U>, options?: Con...'.
                        Types of parameters 'mapper' and 'mapper' are incompatible.
                          Types of parameters 'item' and 'item' are incompatible.
                            Type '{}' is not assignable to type 'Q'.

78     return TagModel.sequelize.query(query, options)
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
79                     .then(data => data.map(d => d.name))
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


 69% building modules 814/815 modules 1 active ...ared/video-image.component.ngfactory.jsnpm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! peertube@1.0.0-beta.16 tsc: `tsc`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the peertube@1.0.0-beta.16 tsc script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
 69% building modules 815/816 modules 1 active ...dations/recommended-videos.component.ts
npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2018-10-02T20_19_16_046Z-debug.log
```

```
# cat /root/.npm/_logs/2018-10-02T20_19_16_046Z-debug.log
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node',
1 verbose cli   '/var/www/peertube/PeerTube/node_modules/.bin/npm',
1 verbose cli   'run',
1 verbose cli   'tsc' ]
2 info using npm@6.4.1
3 info using node@v8.12.0
4 verbose run-script [ 'pretsc', 'tsc', 'posttsc' ]
5 info lifecycle peertube@1.0.0-beta.16~pretsc: peertube@1.0.0-beta.16
6 info lifecycle peertube@1.0.0-beta.16~tsc: peertube@1.0.0-beta.16
7 verbose lifecycle peertube@1.0.0-beta.16~tsc: unsafe-perm in lifecycle true
8 verbose lifecycle peertube@1.0.0-beta.16~tsc: PATH: /var/www/peertube/PeerTube/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/var/www/peertube/PeerTube/node_modules/.bin:/usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/var/www/peertube/PeerTube/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
9 verbose lifecycle peertube@1.0.0-beta.16~tsc: CWD: /var/www/peertube/PeerTube
10 silly lifecycle peertube@1.0.0-beta.16~tsc: Args: [ '-c', 'tsc' ]
11 silly lifecycle peertube@1.0.0-beta.16~tsc: Returned: code: 2  signal: null
12 info lifecycle peertube@1.0.0-beta.16~tsc: Failed to exec tsc script
13 verbose stack Error: peertube@1.0.0-beta.16 tsc: `tsc`
13 verbose stack Exit status 2
13 verbose stack     at EventEmitter.<anonymous> (/var/www/peertube/PeerTube/node_modules/npm/node_modules/npm-lifecycle/index.js:301:16)
13 verbose stack     at emitTwo (events.js:126:13)
13 verbose stack     at EventEmitter.emit (events.js:214:7)
13 verbose stack     at ChildProcess.<anonymous> (/var/www/peertube/PeerTube/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at emitTwo (events.js:126:13)
13 verbose stack     at ChildProcess.emit (events.js:214:7)
13 verbose stack     at maybeClose (internal/child_process.js:915:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
14 verbose pkgid peertube@1.0.0-beta.16
15 verbose cwd /var/www/peertube/PeerTube
16 verbose Linux 4.9.0-8-amd64
17 verbose argv "/usr/bin/node" "/var/www/peertube/PeerTube/node_modules/.bin/npm" "run" "tsc"
18 verbose node v8.12.0
19 verbose npm  v6.4.1
20 error code ELIFECYCLE
21 error errno 2
22 error peertube@1.0.0-beta.16 tsc: `tsc`
22 error Exit status 2
23 error Failed at the peertube@1.0.0-beta.16 tsc script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 2, true ]
```

* **PeerTube version or URL:**
peertube@1.0.0-beta.16 develop branch 

* **Browser name/version:**
Not applicable 

* **NodeJS version:**
v8.12.0

* **What is the expected behaviour?**
Expect for build to finish with success

* **What do you see instead?**
Errors above. Also when running `npm run build` `build:client` continues even if `build:server` fails. Perhaps would be better to not build client if build server fails.

* **Browser console log if useful:**
Maybe these links can help to resolve the issue.
- https://github.com/RobinBuschmann/sequelize-typescript/issues/165
- https://stackoverflow.com/questions/39127714/property-catch-does-not-exist-on-type-promisevoid#39128919
- https://stackoverflow.com/questions/51926755/making-an-interface-generic-over-promise-libraries

* **Server log if useful (journalctl or /var/www/peertube/storage/logs):**

**Comment by Chocobozzz** [comment]
`rm -r ./node_modules && rm -r client/node_modules && yarn install --pure-lockfile`

**Comment by FlatEarthTruther** [comment]
Added a script named `installer` to package.json file so to be able to run suggested command as `npm run installer` or `yarn run installer`.

Will execute with force in case when `node_modules` doesn't exist even:

```
rm -rf ./node_modules && rm -rf client/node_modules && yarn install --pure-lockfile
```

This fixed the issue for me and so to make things easier in the future I've decided to add the command to package.json scripts.

I'll not closing the issue until the pull request gets merged into develop branch.
